src/services/factories/tokenList.ts

```diff
@@ -175,18 +176,56 @@ export function getTokensFactory(factoryParams: {
     const tokenDetailsPromises: (Promise<TokenDetails | undefined> | TokenDetails)[] = []
     addressToIdMap.forEach((id, tokenAddress) => {
       // Resolve the details using the config, otherwise fetch the token
-      const token: TokenDetails | undefined | Promise<TokenDetails | undefined> = tokensConfigMap.has(tokenAddress)
-        ? tokensConfigMap.get(tokenAddress)
+      const token: undefined | Promise<TokenDetails | undefined> = tokensConfigMap.has(tokenAddress)
+        ? Promise.resolve(tokensConfigMap.get(tokenAddress))
         : _fetchToken(networkId, id, tokenAddress)
 
       if (token) {
+        // Add a label for convenience
+        token.then((token) => {
+          if (token) {
+            token.label = safeTokenName(token)
+          }
+          return token
+        })
```

Comment:
As we don't await this specific promise, it could be a race condition
It won't be a race condition in our case because this microtask is scheduled before `await Promise.all`, but still a bad approach.